### PR TITLE
Fix socket type confuse

### PR DESCRIPTION
**Pull request by dark-lbp**
Fix socket_type confuse, we should convert socket type, level and opt to host platform based value.

### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [x] Essential comments are added.
- [x] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [x] Tests will be added after some discussion and review.

### Changelog?

- [ ] This PR doesn't need to update Changelog.
- [x] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

### CODE REVIEW

**Comment by xwings** [comment]
there we go. seems ok now.

let me check the code one more time and will merge

thanks !

File: qiling/os/posix/syscall/socket.py [hunk 1]
```diff
@@ -161,11 +225,59 @@ def ql_syscall_setsockopt(ql, sockfd, level, optname, optval_addr, optlen, *args
...
+                else:
+                    emu_opt_name = socket_option_mapping(emu_opt, ql.archtype)
+
+                # Fix for mips
+                if ql.archtype == QL_ARCH.MIPS:
+                    if emu_opt_name.endswith("_NEW") or emu_opt_name.endswith("_OLD"):
+                        emu_opt_name = emu_opt_name[:-4]
+
+                optname = getattr(socket, emu_opt_name)
+                ql.log.info("Convert emu_optname {}:{} to host platform based optname {}:{}".format(
```
Comment (xwings) [inline qiling/os/posix/syscall/socket.py:260]: this one should be ql.log.debug too

File: qiling/os/posix/syscall/socket.py [hunk 2]
```diff
@@ -161,11 +225,59 @@ def ql_syscall_setsockopt(ql, sockfd, level, optname, optval_addr, optlen, *args
         ql.os.fd[sockfd].setsockopt(level, optname, None, optlen)
     else:
         try:
+            try:
+                emu_level = level
+                emu_level_name = socket_level_mapping(emu_level, ql.archtype)
+                level = getattr(socket, emu_level_name)
+                ql.log.info("Convert emu_level {}:{} to host platform based level {}:{}".format(
```
Comment (xwings) [inline qiling/os/posix/syscall/socket.py:232]: this one should be ql.log.debug too

**Comment by dark-lbp** [comment]
Really sorry for that, opened the fourth pr #882 .

**Pull request closed without merging**